tests/test_hardware_shapes.py

```diff
@@ -0,0 +1,190 @@
+import re
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+from service_capacity_modeling.hardware import shapes
+
+
+def test_r6id() -> None:
+    r6id_24xl = shapes.region("us-east-1").instances["r6id.24xlarge"]
+    assert r6id_24xl is not None
+    assert r6id_24xl.cpu == 96
+
+
+def get_instance_families() -> Dict[str, List[Tuple[str, str]]]:
+    """Extract all instance families from available instances."""
+    region = shapes.region("us-east-1")
+    families: Dict[str, List[Tuple[str, str]]] = {}
+    for instance_name in region.instances:
+        match = re.match(r"([a-z]+\d+[a-z]*)\.(\d+xl|[a-z]+)", instance_name)
+        if match:
+            family, size = match.groups()
```

Comment:
nit: there are already helpers on the Instance objects themselves to get the family and shape from the instance name - this is a little fragile to if we change the names in the keys (which are unstructured)